driver-sync/src/test/functional/com/mongodb/client/AbstractTransactionsTest.java

```diff
@@ -344,6 +244,128 @@ public void shouldPassAllOutcomes() {
+                assertFalse(String.format("Expected error code '%s' but none thrown for operation %s",
+                        getErrorCodeNameField(expectedResult), operationName), hasErrorCodeNameField(expectedResult));
+            } catch (RuntimeException e) {
+                boolean passedAssertion = false;
+                if (hasErrorLabelsContainField(expectedResult)) {
+                    if (e instanceof MongoException) {
+                        MongoException mongoException = (MongoException) e;
+                        for (String curErrorLabel : getErrorLabelsContainField(expectedResult)) {
+                            assertTrue(String.format("Expected error label '%s but found labels '%s' for operation %s",
+                                    curErrorLabel, mongoException.getErrorLabels(), operationName),
+                                    mongoException.hasErrorLabel(curErrorLabel));
+                        }
+                        passedAssertion = true;
+                    }
+                }
+                if (hasErrorLabelsOmitField(expectedResult)) {
+                    if (e instanceof MongoException) {
+                        MongoException mongoException = (MongoException) e;
+                        for (String curErrorLabel : getErrorLabelsOmitField(expectedResult)) {
+                            assertFalse(String.format("Expected error label '%s to be omitted but found labels '%s' for operation %s",
+                                    curErrorLabel, mongoException.getErrorLabels(), operationName),
+                                    mongoException.hasErrorLabel(curErrorLabel));
+                        }
+                        passedAssertion = true;
+                    }
+                }
+                if (hasErrorContainsField(expectedResult)) {
+                    String expectedError = getErrorContainsField(expectedResult);
+                    assertTrue(String.format("Expected '%s' but got '%s' for operation %s", expectedError, e.getMessage(),
+                            operationName), e.getMessage().toLowerCase().contains(expectedError.toLowerCase()));
+                    passedAssertion = true;
+                }
+                if (hasErrorCodeNameField(expectedResult)) {
+                    String expectedErrorCodeName = getErrorCodeNameField(expectedResult);
+                    if (e instanceof MongoCommandException) {
+                        assertEquals(expectedErrorCodeName, ((MongoCommandException) e).getErrorCodeName());
+                        passedAssertion = true;
+                    } else if (e instanceof MongoWriteConcernException) {
+                        assertEquals(expectedErrorCodeName, ((MongoWriteConcernException) e).getWriteConcernError().getCodeName());
+                        passedAssertion = true;
+                    }
+                }
+                if (!passedAssertion || throwExceptions) {
```

Comment:
In the recursive call case when used from withTransaction, throwExceptions will be true, because we want withTransaction closure to propagate exceptions to withTransaction method.